### PR TITLE
Add code coverage tests for notebookUtils

### PR DESCRIPTION
**Pull request by corivera**
Code coverage for notebookUtils.ts is 100% with these tests. Also removed some functions that weren't being used.

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts [hunk 1]
```diff
@@ -3,86 +3,184 @@
...
+		}]);
 	});
 
-	test('Should extract server and database name with other parentheses', async function (): Promise<void> {
-		let serverName = getServerFromFormattedAttachToName('serv()erName (databaseName)');
-		let databaseName = getDatabaseFromFormattedAttachToName('serv()erName (databaseName)');
-		assert.equal(serverName, 'serv()erName');
-		assert.equal(databaseName, 'databaseName');
+	test('getHostAndPortFromEndpoint Test', async function (): Promise<void> {
+		let result = getHostAndPortFromEndpoint('https://localhost:1433');
```
Comment (chlafreniere) [inline src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: just for sanity can we include some ports that aren't 4 digits long as well?

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts [hunk 2]
```diff
@@ -3,86 +3,184 @@
...
 
-	test('Should format server name as empty string when server/database are undefined', async function (): Promise<void> {
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = undefined;
-		connProfile.databaseName = undefined;
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, '');
+	test('tryMatchCellMagic Test', async function (): Promise<void> {
+		let result = tryMatchCellMagic(undefined);
```
Comment (chlafreniere) [inline src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: how about a couple of tests with multiline content?

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts [hunk 3]
```diff
@@ -3,86 +3,184 @@
...
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = 'serverName';
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, 'serverName');
+	test('getProvidersForFileName Test', async function (): Promise<void> {
+		setupMockNotebookService();
+
+		let result = getProvidersForFileName('', mockNotebookService.object);
+		assert.deepStrictEqual(result, [defaultTestProvider]);
```
Comment (chlafreniere) [inline src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: are we positive that deepStrictEqual will return true only if 2 arrays are have identical contents? just sanity checking 😄

Comment (corivera) [reply src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: Yes. Changing the contents of the expected array here fails the test.

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts (hunk 2 above)
Comment (corivera) [reply src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: Done

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts (hunk 1 above)
Comment (corivera) [reply src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: Done

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts [hunk 4]
```diff
@@ -3,86 +3,187 @@
...
+		result = tryMatchCellMagic('%%sql');
+		assert.strictEqual(result, 'sql');
+
+		result = tryMatchCellMagic('%%sql\nselect @@VERSION\nselect * from TestTable');
+		assert.strictEqual(result, 'sql');
+
+		result = tryMatchCellMagic('%%');
+		assert.strictEqual(result, null);
+
+		result = tryMatchCellMagic('%% sql');
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: What about multiple magic's in a single input?

File: src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts [hunk 5]
```diff
@@ -3,86 +3,187 @@
...
-		let serverName = getServerFromFormattedAttachToName('serverName');
-		let databaseName = getDatabaseFromFormattedAttachToName('serverName');
-		assert.equal(serverName, 'serverName');
-		assert.equal(databaseName, '');
+	test('asyncForEach Test', async function (): Promise<void> {
+		let totalResult = 0;
+		await asyncForEach([1, 2, 3, 4], async (value) => {
+			totalResult += value;
+		});
+		assert.strictEqual(totalResult, 10);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts]: undefined and empty array would be good to test here too

**Pull request merged**